Microsoft.DotNet.Interactive.PowerShell/PowerShellKernel.cs

```diff
@@ -0,0 +1,274 @@
+        }
+
+        #region Overrides
+
+        protected override Task HandleAsync(
+            IKernelCommand command,
+            KernelInvocationContext context)
+        {
+            if (command is KernelCommandBase kb)
+            {
+                if (kb.Handler == null)
+                {
+                    switch (command)
+                    {
+                        case SubmitCode submitCode:
+                            submitCode.Handler = (_, invocationContext) =>
+                            {
+                                HandleSubmitCode(submitCode, context);
+                                return Task.CompletedTask;
+                            };
+                            break;
+
+                        case RequestCompletion requestCompletion:
+                            requestCompletion.Handler = (_, invocationContext) =>
+                            {
+                                HandleRequestCompletion(requestCompletion, invocationContext);
+                                return Task.CompletedTask;
+                            };
+                            break;
+
+                        case CancelCurrentCommand interruptExecution:
+                            interruptExecution.Handler = (_, invocationContext) =>
+                            {
+                                HandleCancelCurrentCommand(interruptExecution, invocationContext);
+                                return Task.CompletedTask;
+                            };
+                            break;
+                    }
+                }
+            }
+
+            return Task.CompletedTask;
+        }
+
+        #endregion
+
+        #region Handlers
+
+        private void HandleSubmitCode(
+                SubmitCode submitCode,
+                KernelInvocationContext context)
+        {
+            CancellationTokenSource cancellationSource;
+            lock (_cancellationSourceLock)
+            {
+                cancellationSource = _cancellationSource;
+            }
+
+            // Acknowledge that we received the request.
+            var codeSubmissionReceived = new CodeSubmissionReceived(submitCode);
+            context.Publish(codeSubmissionReceived);
+
+            // Test is the code we got is actually able to run.
+            var code = submitCode.Code;
+            if (IsCompleteSubmission(code, out ParseError[] parseErrors))
+            {
+                context.Publish(new CompleteCodeSubmissionReceived(submitCode));
+            }
+            else
+            {
+                context.Publish(new IncompleteCodeSubmissionReceived(submitCode));
+            }
+
+            // If there were parse errors, publish them and return early.
+            if (parseErrors.Length > 0)
+            {
+                foreach (ParseError parseError in parseErrors)
+                {
+                    context.Publish(new CommandFailed(null, submitCode, message: parseError.ToString()));
```

Comment:
this should  be 
```context.Fail(string.Join("\n", parseErrors.Select(pe => pe.ToString()));```

a command cannot Fail more than once.

Reply:
changed.